common/src/main/java/io/druid/common/config/Log4jShutdown.java

```diff
@@ -95,21 +94,19 @@ public void start()
   @Override
   public void stop()
   {
-    if (callbacksRun.get()) {
-      return;
-    }
-    if (!state.compareAndSet(State.STARTED, State.STOPPED)) {
+    if (!state.compareAndSet(State.STARTED, State.STOPPING)) {
```

Comment:
Note that https://github.com/metamx/java-util/blob/master/src/main/java/com/metamx/common/lifecycle/Lifecycle.java#L267 does synchronize, so it may be sufficient to simply state in a comment on the stop method that synchronization is assumed to be performed in the caller